File: javascript/8941.js
const input = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n");
const mass = { C: 12.01, H: 1.008, O: 16.0, N: 14.01 };

input.slice(1).forEach((line) => {
  const sum = [...line.matchAll(/([CHON])(\d*)/g)].reduce(
    (acc, [_, atom, num]) => mass[atom] * parseInt(num || 1) + acc,
    0
  );

  console.log(sum.toFixed(3));
});

File: python/8941.py
import re

mass = {"C": 12.01, "H": 1.008, "O": 16.0, "N": 14.01}

for _ in range(int(input())):
    mole_sum = sum(
        [
            mass[atom] * (int(num) if num else 1)
            for atom, num in re.findall(r"([CHON])(\d*)", input())
        ]
    )

    print(f"{mole_sum:.3f}")
